libs/application/template-api-modules/src/lib/modules/templates/energy-funds/energy-funds.service.ts

```diff
@@ -0,0 +1,141 @@
+      throw new TemplateApiError(
+        {
+          title: coreErrorMessages.electricVehicleListEmptyOwner,
+          summary: coreErrorMessages.electricVehicleListEmptyOwner,
+        },
+        400,
+      )
+    }
+
+    return await Promise.all(
+      onlyElectricVehicles?.map(async (vehicle) => {
+        let vehicleGrantPriceAmount: number | undefined
+        let vehicleGrantItemCode: string | undefined
+        let hasReceivedSubsidy: boolean | undefined
+
+        // Only validate if fewer than 5 items
+        if (onlyElectricVehicles.length < 5) {
+          const vehicleGrant = await this.getVehicleGrant(auth, vehicle)
+          vehicleGrantPriceAmount = vehicleGrant?.priceAmount
+          vehicleGrantItemCode = vehicleGrant?.itemCode
+
+          // Get subsidy status
+          hasReceivedSubsidy =
+            await this.energyFundsClientService.checkVehicleSubsidyAvilability(
+              auth,
+              vehicle.vin || '',
+            )
+        }
+
+        return {
+          permno: vehicle.permno,
+          vin: vehicle.vin,
+          make: vehicle.make,
+          color: vehicle.color,
+          role: vehicle.role,
+          firstRegistrationDate: vehicle.firstRegistrationDate,
+          newRegistrationDate: vehicle.newRegistrationDate,
+          fuelCode: vehicle.fuelCode,
+          vehicleRegistrationCode: vehicle.vehicleRegistrationCode,
+          importCode: vehicle.importCode,
+          vehicleGrant: vehicleGrantPriceAmount,
+          vehicleGrantItemCode: vehicleGrantItemCode,
+          hasReceivedSubsidy: hasReceivedSubsidy,
+        }
+      }),
+    )
+  }
+
+  async submitApplication({
+    auth,
+    application,
+  }: TemplateApiModuleActionProps): Promise<void> {
+    const applicationAnswers = application.answers as EnergyFundsAnswers
+    const currentVehicleList = application.externalData?.currentVehicles
+      ?.data as Array<VehiclesCurrentVehicle>
+    const currentvehicleDetails = currentVehicleList.find(
+      (x) => x.permno === applicationAnswers.selectVehicle.plate,
+    )
+
+    if (currentvehicleDetails) {
+      const answers = {
+        nationalId: auth.nationalId,
+        vIN: applicationAnswers?.selectVehicle.vin,
+        carNumber: applicationAnswers?.selectVehicle.plate,
+        carType: currentvehicleDetails.make || '',
+        itemcode: currentvehicleDetails.vehicleGrantItemCode || '',
+        purchasePrice:
+          (applicationAnswers?.vehicleDetails.price &&
+            parseInt(applicationAnswers?.vehicleDetails.price)) ||
+          0,
+        registrationDate: format(
+          new Date(currentvehicleDetails.firstRegistrationDate || ''),
+          'yyyy-MM-dd',
+        ),
+        subsidyAmount: currentvehicleDetails.vehicleGrant || 0,
+      }
+
+      await this.energyFundsClientService.submitEnergyFundsApplication(auth, {
+        subsidyInput: answers,
+      })
+    } else {
+      throw new TemplateApiError(
```

Comment:
This is a thrown validation error but the user has no recourse as it has no further information on how to react to the error as it will only read "Umsókn mistókst".  Should at least provide the user with some direction on how to proceed. 
This would be when the user has not selected any vehicles ?

Reply:
I realize this case couldn't ever happen as the user would get stuck in the dataProvider stage if they have no vehicles that match the application requirements. I removed this check and just added a null check on the variables in the answer object